dashboard/modules/log/log_agent.py

```diff
@@ -370,6 +374,38 @@ async def _find_task_log_offsets(
 
         return start_offset, end_offset
 
+    @classmethod
+    def _resolve_filename(cls, root_log_dir: Path, filename: str) -> Path:
+        """
+        Resolves the file path relative to the root log directory.
+
+        Args:
+            root_log_dir: Root log directory.
+            filename: File path relative to the root log directory.
+
+        Raises:
+            FileNotFoundError: If the file path is invalid.
+
+        Returns:
+            The absolute file path resolved from the root log directory.
+        """
+        if not Path(filename).is_absolute():
+            filepath = root_log_dir / filename
+        else:
+            filepath = Path(filename)
+
+        filepath = filepath.resolve()
+
+        if not filepath.is_file():
+            raise FileNotFoundError(f"A file is not found at: {filepath}")
+
+        try:
+            filepath.relative_to(root_log_dir)
+        except ValueError as e:
+            raise FileNotFoundError(f"{filepath} not in {root_log_dir}: {e}")
```

Comment:
Out of curiosity, any reason we're going the exception throwing path vs `is_relative_to`?

Reply:
I had the same comment (@rickyyx feel free to make a follow up PR to change this, but it is cosmetic so I'm merging the PR which is an important fix)

Reply:
oh - I think that's only available in py3.9?
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.is_relative_to

Reply:
Oooo, very good catch. Thanks!